tests/e2e/performance/test_pvc_multi_snapshot_performance.py

```diff
@@ -0,0 +1,157 @@
+
+        self.interface = interface_iterate
+        self.sc_obj = storageclass_factory(self.interface)
+
+        self.pvc_obj = pvc_factory(
+            interface=self.interface,
+            size=pvc_size,
+            status=constants.STATUS_BOUND
+        )
+
+        self.pod_obj = pod_factory(
+            interface=self.interface,
+            pvc=self.pvc_obj,
+            status=constants.STATUS_RUNNING
+        )
+
+        # Calculating the file size as 80% of the PVC size
+        filesize = self.pvc_obj.size * 0.80
+        # Change the file size to MB for the FIO function
+        file_size = f'{int(filesize * constants.GB2MB)}M'
+        file_name = self.pod_obj.name
+
+        log.info(
+            f'Total capacity size is : {ceph_capacity} GiB, '
+            f'Going to use {need_capacity} GiB, '
+            f'With {num_of_snaps} Snapshots to {pvc_size} GiB PVC. '
+            f'File size to be written is : {file_size} '
+            f'with the name of {file_name}'
+        )
+
+        snap_yaml = constants.CSI_RBD_SNAPSHOT_YAML
+        if self.interface == constants.CEPHFILESYSTEM:
+            snap_yaml = constants.CSI_CEPHFS_SNAPSHOT_YAML
+
+        for test_num in range(1, (num_of_snaps + 1)):
+
+            test_results = {
+                'test_num': test_num,
+                'create': {'time': None, 'speed': None},
+            }
+
+            # Going to run only write IO to fill the PVC for the snapshot - will run 1M BS for 3 Min. 1M
+            log.info(f"Running IO on pod {self.pod_obj.name} - Test number {test_num}")
+            self.pod_obj.fillup_fs(size=file_size, fio_filename=file_name)
+
+            # Wait for fio to finish
+            fio_result = self.pod_obj.get_fio_results()
+            err_count = fio_result.get('jobs')[0].get('error')
+            assert err_count == 0, (
+                f"IO error on pod {self.pod_obj.name}. "
+                f"FIO result: {fio_result}"
+            )
+            log.info('IO on the PVC Finished')
+
+            # Take a snapshot of the PVC and measure the time of creation.
+            snap_name = self.pvc_obj.name.replace(
+                'pvc-test', f'snapshot-test{test_num}'
+            )
+            log.info(f'Taking snapshot of the PVC {snap_name}')
+            snap_obj = pvc.create_pvc_snapshot(
+                self.pvc_obj.name,
+                snap_yaml,
+                snap_name,
+                helpers.default_volumesnapshotclass(self.interface).name,
+            )
+            snap_obj.ocp.wait_for_resource(
+                condition='true', resource_name=snap_obj.name,
+                column=constants.STATUS_READYTOUSE, timeout=60
+            )
+
+            snap_con_name = snap_obj.ocp.get(
+                resource_name=snap_name,
+                out_yaml_format=True
+            )["status"]["boundVolumeSnapshotContentName"]
+            log.info(f'Snap content name is {snap_con_name}')
```

Comment:
Adding 'test_num' will be helpful for analysis

Reply:
add.